src/main/asciidoc/java/parsetools.adoc

```diff
@@ -47,4 +47,164 @@ RecordParser.newFixed(4, h -> {
\ No newline at end of file
+For more details, check out the `link:../../apidocs/io/vertx/core/parsetools/RecordParser.html[RecordParser]` class.
+
+== Json Parser
+
+You can easily parse JSON structures but that requires to provide the JSON string at once, but it
+may not be convenient when you need to parse very large structures.
+
+The non-blocking JSON parser is an event driven parser able to deal with very large structures.
+It transforms a sequence of input buffer to a sequence of JSON parse events.
+
+[source, java]
+----
+JsonParser parser = JsonParser.newParser();
+
+// Set handlers for various events
+parser.handler(event -> {
+  switch (event.type()) {
+    case START_OBJECT:
+      // Start an objet
+      break;
+    case END_OBJECT:
+      // End an objet
+      break;
+    case START_ARRAY:
+      // Start an array
+      break;
+    case END_ARRAY:
+      // End an array
+      break;
+    case VALUE:
+      // Handle a value
+      String field = event.field();
+      if (field != null) {
+        // In an object
+      } else {
+        // In an array or top level
+        if (event.isString()) {
+
+        } else {
+          // ...
+        }
+      }
+      break;
+  }
+});
+----
+
+The parser is non-blocking and emitted events are driven by the input buffers
+
+[source, java]
+----
+JsonParser parser = JsonParser.newParser();
+
+// start array event
+// start object event
+// "firstName":"Bob" event
+parser.handle(Buffer.buffer("[{\"firstName\":\"Bob\"},"));
+
+// "lastName":"Morane" event
+// end object event
+parser.handle(Buffer.buffer("\"lastName\":\"Morane\")},"));
+
+// start object event
+// "firstName":"Luke" event
+// "lastName":"Lucky" event
+// end object event
+parser.handle(Buffer.buffer("{\"firstName\":\"Luke\",\"lastName\":\"Lucky\")}"));
+
+// end array event
+parser.handle(Buffer.buffer("]"));
+
+// Always call end
+parser.end();
+----
+
+Event driven parsing provides more control but comes at the price of flexibility. The parser allows you
+to handle JSON structures as values when it is desired:
+
+[source, java]
+----
+JsonParser parser = JsonParser.newParser();
+
+parser.objectValueMode();
+
+parser.handler(event -> {
+  switch (event.type()) {
+    case START_ARRAY:
+      // Start the array
+      break;
+    case END_ARRAY:
+      // End the array
+      break;
+    case VALUE:
+      // Handle each object
+      break;
+  }
+});
+
+parser.handle(Buffer.buffer("[{\"firstName\":\"Bob\"},\"lastName\":\"Morane\"),...]"));
+parser.end();
+----
+
+The value mode can be set and unset during the parsing allowing you to switch between fine grained
+events or JSON object/array value events.
+
+[source, java]
+----
+JsonParser parser = JsonParser.newParser();
+
+parser.handler(event -> {
+  // Start the object
+
+  switch (event.type()) {
+    case START_OBJECT:
+      // Set object value mode to handle each entry, from now on the parser won't emit start object events
```

Comment:
IIUC, when object value mode is enabled, all nested objects/arrays are parsed at once?

Reply:
I just reviewed the parser impl, it seems to be the case. Could you add tests for nested objects/arrays?